project/src/test/java/com/google/sps/UserProfileEventsTest.java

```diff
@@ -0,0 +1,181 @@
+import org.mockito.Mockito;
+import org.springframework.mock.web.MockServletContext;
+
+/** Unit tests for updates and retrievals for a user's corresponding events. */
+@RunWith(JUnit4.class)
+public class UserProfileEventsTest {
+  private HttpServletRequest request;
+  private HttpServletResponse response;
+  private StringWriter stringWriter;
+  private PrintWriter printWriter;
+  private UserProfileEventsServlet profileEventsServlet;
+  private static final LocalServiceTestHelper authenticationHelper =
+      new LocalServiceTestHelper(new LocalUserServiceTestConfig());
+  private static final String PARAMETER_EVENT_TYPE = "event-type";
+  private static final String VOLUNTEERING = "volunteering";
+  private static final String HOST_NAME = "Bob Smith";
+  private static final String EMAIL = "bobsmith@example.com";
+  private static final String INVALID_EVENT_TYPE = "all";
+
+  @Before
+  public void setUp() throws Exception {
+    MockServletContext mockServletContext = new MockServletContext();
+    new SpannerClient().contextInitialized(new ServletContextEvent(mockServletContext));
+    SpannerTestTasks.setup();
+
+    authenticationHelper.setUp();
+
+    profileEventsServlet = new UserProfileEventsServlet();
+    
+    request = Mockito.mock(HttpServletRequest.class);
+    response = Mockito.mock(HttpServletResponse.class);
+    stringWriter = new StringWriter();
+    printWriter = new PrintWriter(stringWriter);
+    Mockito.when(response.getWriter()).thenReturn(printWriter);
+  }
+
+  @After
+  public void tearDown() {
+    SpannerTestTasks.cleanup();
+    authenticationHelper.tearDown();
+  }
+
+  @Test
+  public void getUserEventsHosting() throws IOException {
+    
+  }
+
+  @Test
+  public void getUserEventsParticipating() throws IOException {
+    
+  }
+
+  @Test
+  public void getUserEventsVolunteering_noVolunteerSignupsForUser() throws IOException {
+    String userEmail = "test@gmail.com";
+    User user = TestUtils.newUserWithEmail(userEmail);
+    SpannerTasks.insertOrUpdateUser(user);
+
+    User host = new User.Builder(HOST_NAME, EMAIL).build();
+    SpannerTasks.insertOrUpdateUser(host);
+    Event event = TestUtils.newEventWithHost(host);
+    SpannerTasks.insertorUpdateEvent(event);
+
+    VolunteeringOpportunity opportunity =
+    TestUtils.newVolunteeringOpportunityWithEventId(event.getId());
+    SpannerTasks.insertVolunteeringOpportunity(opportunity);
+
+    authenticationHelper
+        .setEnvIsLoggedIn(true)
+        .setEnvEmail(userEmail)
+        .setEnvAuthDomain("gmail.com");
+    Mockito.when(request.getParameter(PARAMETER_EVENT_TYPE)).thenReturn(VOLUNTEERING);
+
+    profileEventsServlet.doGet(request, response);
+
+    Assert.assertEquals(
+        CommonUtils.convertToJson(new HashSet<>()).trim(),
+        stringWriter.toString().trim());
+  }
+
+  @Test
+  public void getUserEventsVolunteering_nonzeroVolunteerSignupsForUser() throws IOException {
+    String userEmail = "test@gmail.com";
```

Comment:
Move test@gmail.com to a constant so you can use the same var in all the tests?

Reply:
Done.